### PR TITLE
Add 'shell.nix'

### PR DESCRIPTION
**Pull request by nkaretnikov**

### CODE REVIEW

**Comment by ndmitchell** [comment]
This seems fine to include if it's useful. My only question is why haven't I previously got patches for all my other packages, which are generally more popular than bake? Do most other packages go upstream? Is this a stopgap before Bake is included as a Nix supplied package?

**Comment by nkaretnikov** [comment]
> This seems fine to include if it's useful. My only question is why
> haven't I previously got patches for all my other packages, which are
> generally more popular than bake? Do most other packages go upstream?
> Is this a stopgap before Bake is included as a Nix supplied package?

All the Haskell Nix recipes are (semi-)automatically generated from
Hackage (warning: that's a big file):

https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/development/haskell-modules/hackage-packages.nix

That's probably why you haven't received any patches of this sort.  Note
that Bake is also there.  But Bake is also special in a sense that you
suggest to use the git checkout, and that's why I opened this pull
request.  The `shell.nix` file is for folks who first clone your git
repo.  Then, it's just a matter of running `nix-shell --pure` (which
fetches all the deps and drops you into an environment where all of them
are available) and the usual `cabal configure && cabal build`.

It's not difficult to create a `shell.nix` file.  It took me a minute or
so.  I just ran `cabal2nix --shell --sha256=0 bake.cabal > shell.nix`
and edited some things manually.  I, however, forgot to include `git`,
so it's not possible to, say, commit things inside the nix shell.  But
I'll fix that if you think that this patch is useful.

To summarize: no, it's not a stopgap, it just makes things a bit easier
for Nix users.

**Comment by nkaretnikov** [comment]
Oh, if you expect folks to pull from the git repo for a while instead of using Hackage, I can talk to the NixOS people about changing the script they have to point to this repo.

Also, in order to depend on Bake as a library, it's necessary to provide a `default.nix` file, which uses a slightly different syntax. I didn't submit it yesterday because I was just interested in building the project and toying around a bit.

Sorry for the wall of text, I hope it's not too confusing. Let me know how you feel about this, and I make sure to do everything right.

**Comment by ndmitchell** [comment]
Thanks for all the info, that all makes sense, much appreciated. I hope to move to real releases shortly - it's nearly at that point already.
